### ISSUE TITLE
GraphQL & Kotlin: Schema Creator for Queries with Collections does not consider the NotNull/Nullable

### ISSUE THREAD

**Issue by vladi-bnv** [smallrye/smallrye-graphql#1992]
The SchemaCreator seems to ignore the Optional/NotOptional specifications for Kotlin's query methods 
Here is an example:
```
data class Output(val someNullableData: List<String?>?,
                  val someData: List<String>)

data class Input @JsonbCreator constructor(
    val someNullableData: List<String?>?,
    val someData: List<String>,
)


@ApplicationScoped
@GraphQLApi
class KotlinGraphQLQueries {
    // Input and output are nullable
    @Query
    fun queryOne(input: List<Input?>?): List<Output?>? {
        return listOf()
    }
    // List elements are nullable, lists are not nullable
    @Query
    fun queryTwo(input: List<Input?>): List<Output> {
        return listOf()
    }
    @Query
    fun queryThree(input: List<Input>): List<Output?> {
        return listOf()
    }
    // Nothing is nullable
    @Query
    fun queryFour(input: List<Input>): List<Output> {
        return listOf()
    }
}
```

Schema Creator produces:
```
type Output {
  someData: [String]! # Wrong: list elem is nonnull
  someNullableData: [String]
}
input InputInput {
  someData: [String]! # Wrong: list elem is nonnull
  someNullableData: [String]
}


type Query {  
  queryOne(input: [InputInput]): [Output]
  queryTwo(input: [InputInput]!): [Output]! # Wrong: output list elem is nonnull
  queryThree(input: [InputInput]!): [Output]! # Wrong: input list elem is nonnull
  queryFour(input: [InputInput]!): [Output]! # Wrong: input/output list elem are nonnull
}
```

The comments for the list elements don't seem to be working. At first glance, I also couldn't find them in the Jandex index.

**Comment by jmartisk** [comment]
Yeah using Jandex for deducing the nullability of Kotlin types is quite tricky because of how the Kotlin compiler stores the metadata all over the place. We've only recently added some basic implementation of it (see https://github.com/smallrye/smallrye-graphql/blob/2.6.1/common/schema-builder/src/main/java/io/smallrye/graphql/schema/creator/OperationCreator.java#L118) and there are definitely cases where this doesn't work properly. If you want, I invite you to dig into it (looking at the code I linked might be a good start), or I can try it, but not earlier than in January. Happy holidays :)

**Comment by vladi-bnv** [comment]
Thank you, you are welcome to invite me!

Merry Christmas

**Comment by vladi-bnv** [comment]
I have tried to fix the error, it doesn't look bad at first glance. Can I contribute a PR? Or should I attach a patch?

**Comment by jmartisk** [comment]
Submit a PR please :)

**Comment by vladi-bnv** [comment]
Please take a look on my PR: https://github.com/smallrye/smallrye-graphql/pull/2008

**Comment by vladi-bnv** [comment]
In which released version could the fix be included?

**Comment by jmartisk** [comment]
Sorry for missing your question @vladi-bnv 
This has been fixed since 2.8.0
